src/controllers/dynakube/dynakube_controller.go

```diff
@@ -173,29 +169,37 @@ func (controller *Controller) createDynakubeMapper(ctx context.Context, dynakube
 	return &dkMapper
 }
 
-func (controller *Controller) reconcileIstio(dynakube *dynatracev1beta1.DynaKube) bool {
-	updated := false
-
-	if dynakube.Spec.EnableIstio {
-		communicationHosts := connectioninfo.GetCommunicationHosts(dynakube)
-
-		ic, err := istioclientset.NewForConfig(controller.config)
-
-		if err != nil {
-			log.Error(err, "failed to initialize istio client")
-			return false
-		}
+func (controller *Controller) setupIstio(ctx context.Context, dynakube *dynatracev1beta1.DynaKube) (*istio.Reconciler, error) {
+	if !dynakube.Spec.EnableIstio {
+		return nil, nil
+	}
+	istioClient, err := controller.istioClientBuilder(controller.config, controller.scheme, dynakube.Namespace)
+	if err != nil {
+		return nil, errors.WithMessage(err, "failed to initialize istio client")
+	}
 
-		updated, err = istio.NewReconciler(controller.config, controller.scheme, ic).Reconcile(dynakube, communicationHosts)
-		if err != nil {
-			// If there are errors log them, but move on.
-			log.Info("istio failed to reconcile objects", "error", err)
-		}
+	isInstalled, err := istioClient.CheckIstioInstalled()
+	if err != nil {
+		return nil, errors.WithMessage(err, "failed to initialize istio client")
+	} else if !isInstalled {
+		return nil, errors.New("istio not installed yet is enabled, aborting reconciliation, check configuration")
 	}
-	return updated
+	istioReconciler := istio.NewReconciler(istioClient)
+	err = istioReconciler.ReconcileAPIUrl(ctx, dynakube)
+	if err != nil {
+		return nil, errors.WithMessage(err, "failed to reconcile istio objects for API url")
+	}
+	log.Info("reconciled istio objects for API url")
```

Comment:
why not putting this log into the reconciler itself?

Reply:
done